### ISSUE TITLE
total_daily_energy sensor doesn't reset the daily watt consumation at midnight anymore

### ISSUE THREAD

**Issue by h0jeZvgoxFepBQ2C** [esphome/issues#1438]
**Operating environment/Installation (Hass.io/Docker/pip/etc.):**

Native Linux install via pip

**ESP (ESP32/ESP8266, Board/Sonoff):**

Sonoff POW with ESP8266

**ESPHome version (latest production, beta, dev branch)**

latest pre release 1.15 (.b4?)

**Affected component:**

total_daily_energy sensor

**Description of problem:**
Energy sensor is not resetting anymore after update (see graph).
Usually my esp node resets at 00:00:00 and counts up again, but after my last update (i don't know the old version anymore unfortunately) the reset doesn't work anymore as it seems.

<img width="350" alt="Bildschirmfoto 2020-08-24 um 23 06 33" src="https://user-images.githubusercontent.com/10075/91096867-0649c180-e65f-11ea-83d4-d9292d00899d.png">

**Problem-relevant YAML-configuration entries:**
```yaml
platform: total_daily_energy
name: "${description} Stromverbrauch Heute"
power_id: energy_sensor
id: energy_consumption
```

**Logs (if applicable):**

No special log output available

**Additional information and things you've tried:**

Restarting didn't change anything.

Jfyi: I noticed that I looked at the wrong sensor before, so I changed the title afterwards to the correct one.

**Comment by 7h30n3** [comment]
I had the same issue. 

For me it got solved by uprading from 1.15.0b3 to 1.15.0b4.
I'm using a Gosund SP112 and home assistant to sync time.

**Comment by h0jeZvgoxFepBQ2C** [comment]
Can confirm, it works now on 1.15.0b4! Thanks for fixing it!
